### DISCUSSION TITLE
Is it possible to obtain the file paths of user-selected files in file explorer?

### DISCUSSION THREAD

**Discussion by HuakunShen** [tauri-apps/tauri#7769] [Q&A]
Expectation: 
After a user selects a few files in file explorer, the user can press a hotkey and the tauri app could get the file paths.

Is this possible with Tauri? Or Rust?

I have seen an Electron app doing that, but I don't know how Electron did that as well.

**Comment by FabianLars** [comment]
I never saw something like this before, do you still remember which electron app did that?

**Comment by HuakunShen** [comment]
I tried an app called raycast, (native mac app) it can read the text I highlighted as well as files selected.
At first I thought it may dispatch ctrl+c, then read clipboard, but no content is in clipboard history.

I realize that before the feature is enabled, it asked for accessibility permission on Mac.
I believe some accessibility API is used. Didn't check what exactly is used as I am not familiar with Mac native app.

![image](https://github.com/tauri-apps/tauri/assets/33727687/d0fb921b-8858-42cb-b949-b91cd954961a)

Tauri would be more powerful with these APIs.

**Comment by HuakunShen** [comment]
It seems like the API get be used to access another app's UI. https://developer.apple.com/library/archive/samplecode/UIElementInspector/Introduction/Intro.html
https://stackoverflow.com/questions/6836278/api-for-accessing-ui-elements-in-mac-os-x
